test/specs/integration.spec.js

```diff
@@ -65,125 +62,116 @@ describe('Integration tests', function() {
         cm1.replaceRange('', {line: 0, ch: 10}, {line: 0, ch: 13});
         cm1.replaceRange('', {line: 0, ch: 0},  {line: 0, ch: 3});
       });
+      setTimeout(function() {
+        expect(cm2.getValue()).toEqual('3456789');
+        done();
+      }, 500);
```

Comment:
Eeek.  Not a big a fan of arbitrary setTimeout() calls in tests.  This will slow the test down in the common case, and potentially make the test flaky (e.g. if you're on a slow machine or internet connection and it takes longer than 500ms).

You could maybe do something like:

    cm2.on('change', function() {
      if (cm2.getValue() === '3456789') {
        done();
      }
    });